[gitlab] remove duplicate issue lookup in link flow

dcramer opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/getsentry/sentry-plugins/28)
<!-- Reviewable:end -->

dcramer commented:
this is what I was talking about yesterday -- where the updates to the link_issue flow let us remove this duplicate call

macqueen commented on src/sentry_plugins/github/plugin.py:
```diff
@@ -160,9 +160,10 @@ def link_issue(self, request, group, form_data, **kwargs):
                 client.create_comment(
                     repo=self.get_option('repo', group.project),
-                    issue_id=form_data['issue_id'],
+                    global_issue_id=issue['id'],
```
umm this is the github plugin...

mattrobenolt replied:
lol

dcramer commented:
Damnit. Idk how I fucked that one up. Gonna add some tests.

dcramer commented:
Added some tests and fixed a bunch of minor quirks. Still can probably do more here, but at the very least it'll be hard to break these two in the future.

macqueen commented on tests/gitlab/test_plugin.py:
```diff
@@ -74,3 +71,35 @@ def test_create_issue(self):
         }
+
+    @responses.activate
+    @override_settings(GITHUB_APP_ID='abc', GITHUB_API_SECRET='123')
```
should this be gitlab or github? (also on line 48)
